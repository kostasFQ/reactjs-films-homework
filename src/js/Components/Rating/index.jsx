import React from 'react';
import PropTypes from 'prop-types';
import shorid from 'shortid';
import styles from './rating.scss';
import RatingBox from '../RatingBox';
import whiteStar from '../../../imgs/icons/whiteStar.png';
import blueStar from '../../../imgs/icons/star.png';

const Rating = (props) => {
  const { rating } = props;

  const stars = (color) => {
    const arr = new Array(10);
    arr.fill(color);
    return arr;
  };

  if (rating <= 10 && rating > 0) {
    return (
      <div className={styles.container}>
        <div className={styles.ratingBox}>
          { stars(whiteStar).map(item => <img src={item} alt="star" className={styles.star} key={shorid.generate()} />)
            }
        </div>
        <div className={styles.x} style={{ width: `${rating * 20}px` }}>
          {
              stars(blueStar).map(item => <img src={item} alt="star" className={styles.star} key={shorid.generate()} />)
            }
        </div>
        <RatingBox rating={rating} />
      </div>
    );
  }
  return (<div>Sorry, no rating</div>);
};

export default Rating;

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
};
