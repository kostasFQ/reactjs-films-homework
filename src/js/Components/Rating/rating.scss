.container {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
}

.star {
  height: 20px;
  width: 20px;
  display: inline-block;
}

.x {
  z-index: 1;
  position: absolute;
  display: flex;
  overflow: hidden;

  img {
    flex-shrink: 0;
  }
}

.ratingBox {
  min-width: 200px;
}
