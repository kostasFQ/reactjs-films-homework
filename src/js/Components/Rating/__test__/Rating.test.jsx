import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '../index';

test('test component render (rating is correct)', () => {
  const render = renderer.create(
    <Rating rating={5} />,
  );
  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('div');
});

test('test component render (rating is incorrect)', () => {
  const render = renderer.create(
    <Rating rating={8} />,
  );
  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('div');
});
