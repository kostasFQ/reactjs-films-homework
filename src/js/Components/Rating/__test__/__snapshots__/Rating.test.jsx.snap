// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component render (rating is correct) 1`] = `
<div
  className="container"
>
  <div
    className="ratingBox"
  >
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
  </div>
  <div
    className="x"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
    <img
      alt="star"
      className="star"
      src="test-file-stub"
    />
  </div>
  <div
    className="container"
  >
    5
  </div>
</div>
`;

exports[`test component render (rating is incorrect) 1`] = `
<div>
  Sorry, no rating
</div>
`;
